src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -87,24 +92,39 @@ private void raiseAlert(VulnType currentVT, String evidence, HttpMessage msg, in
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 		long start = System.currentTimeMillis();
-		if (msg.getResponseBody().length() > 0 && 
-				msg.getResponseHeader().isText() &&
-				msg.getRequestHeader().isSecure()){ //No point reporting for non-SSL resources
+		Vector<String> STSOption = msg.getResponseHeader().getHeaders("Strict-Transport-Security");
+		String metaHSTS = getMetaHSTSEvidence(source);
+		
+		if (msg.getRequestHeader().isSecure()){ //No point reporting missing for non-SSL resources
 			//Content available via both HTTPS and HTTP is a separate though related issue
-			Vector<String> STSOption = msg.getResponseHeader().getHeaders("Strict-Transport-Security");
 			if (STSOption == null) { // Header NOT found
 					raiseAlert(VulnType.HSTS_MISSING, null, msg, id);
-					return; //No point continuing
-			} else { 
-				for (String stsHeader : STSOption) {
-					Matcher matcher = BAD_MAX_AGE_PATT.matcher(stsHeader);
-					if (matcher.find()) { 
-						String evidence = matcher.group();
-						raiseAlert(VulnType.HSTS_MAX_AGE_DISABLED, evidence, msg, id);
+			} else if (STSOption.size() > 1){//More than one header found
+				raiseAlert(VulnType.HSTS_MULTIPLE_HEADERS, null, msg, id);
+			} else { //Single HSTS header entry
+					Matcher bad_age_matcher = BAD_MAX_AGE_PATT.matcher(STSOption.get(0));
+					Matcher max_age_matcher = MAX_AGE_PATT.matcher(STSOption.get(0));
+					if (bad_age_matcher.find()) {
+						//Matched BAD_MAX_AGE_PATT, max-age is zero
+						raiseAlert(VulnType.HSTS_MAX_AGE_DISABLED, bad_age_matcher.group(), msg, id);
+					} else if (!max_age_matcher.find()) {
+						//Didn't find a digit value associated with max-age
+						raiseAlert(VulnType.HSTS_MAX_AGE_MISSING, STSOption.get(0), msg, id);
 					}
-				}
 			}
+		} 
+		
+		if (STSOption != null && !STSOption.isEmpty() && !msg.getRequestHeader().isSecure()) {

```

Comment:
Could be "else if" instead of just "if" (also making the isSecure check redundant).